share/completions/tmutil.fish

```diff
@@ -1,61 +1,136 @@
+
+complete -c tmutil -f -n __fish_use_subcommand -a enable -d 'Turn on automatic backups'
+
+complete -c tmutil -f -n __fish_use_subcommand -a enablelocal -d 'Turn on local Time Machine snapshots'
+
+complete -c tmutil -r -n __fish_use_subcommand -a inheritbackup -d 'Claim a machine directory or sparsebundle for use by the current machine'
+
+complete -c tmutil -F -n __fish_use_subcommand -a isexcluded -d 'Determine if a file, directory, or volume are excluded from backups'
+
+complete -c tmutil -r -n __fish_use_subcommand -a latestbackup -d 'Print the path to the latest snapshot'
+
+complete -c tmutil -f -n __fish_use_subcommand -a listlocalsnapshotdates -d 'List the creation dates of all local Time Machine snapshots'
+
+complete -c tmutil -r -n __fish_use_subcommand -a listlocalsnapshots -d 'List local Time Machine snapshots of the specified volume'
+
+complete -c tmutil -f -n __fish_use_subcommand -a listbackups -d 'Print paths for all snapshots'
+
+complete -c tmutil -f -n __fish_use_subcommand -a localsnapshot -d 'Create new local Time Machine snapshot of APFS volume in TM backup'
+
+complete -c tmutil -f -n __fish_use_subcommand -a machinedirectory -d 'Print the path to the current machine directory'
+
+complete -c tmutil -f -n __fish_use_subcommand -a removedestination -d 'Removes a backup destination'
+complete -c tmutil -f -n '__fish_seen_subcommand_from removedestination' -d "$(__destination_ids)"
+
+complete -c tmutil -f -n __fish_use_subcommand -a removeexclusion -d 'Remove an exclusion'
+complete -c tmutil -f -n '__fish_seen_subcommand_from removeexclusion' -s v -d 'Volume exclusion'
+complete -c tmutil -f -n '__fish_seen_subcommand_from removeexclusion' -s p -d 'Path exclusion'
+
+complete -c tmutil -f -n __fish_use_subcommand -a restore -d 'Restore an item'
+complete -c tmutil -r -n '__fish_seen_subcommand_from restore' -s v
+
+complete -c tmutil -r -n __fish_use_subcommand -a setdestination -d 'Set a backup destination'
+complete -c tmutil -f -n '__fish_seen_subcommand_from setdestination' -s a -d 'Add to the list of destinations'
+complete -c tmutil -f -n '__fish_seen_subcommand_from setdestination' -s p -d 'Enter the password at a non-echoing interactive prompt'
+
+complete -c tmutil -f -n __fish_use_subcommand -a snapshot -f -d 'Create new local Time Machine snapshot'
+
+complete -c tmutil -f -n __fish_use_subcommand -a startbackup -d 'Begin a backup if one is not already running'
+complete -c tmutil -f -n '__fish_seen_subcommand_from startbackup' -s a -l auto -d 'Automatic mode'
+complete -c tmutil -f -n '__fish_seen_subcommand_from startbackup' -s b -l block -d 'Block until finished'
+complete -c tmutil -f -n '__fish_seen_subcommand_from startbackup' -s r -l rotation -d 'Automatic rotation'
+complete -c tmutil -f -n '__fish_seen_subcommand_from startbackup' -s d -l destination -r -a "$(__destination_ids)" -d 'Backup destination'
```

Comment:
This:

```fish
"$(...)"
```

is a command substitution that will be expanded when this is sourced.

What you want is to use `-a`'s ability to do that when the argument is needed:

```fish
-a "(__destination_ids)"
```

(or `'$(__destination_ids)'` - the important part is that the command substitution is quoted so it doesn't run when the `complete` command is run, but when the `-a` is evaluated)

Reply:
That's brilliant! Essentially avoid expanding the command substitution within the quotes and it gets run later.

This works as expected in my testing.